### PR TITLE
add button for collapsing/expanding all nested attributes

### PR DESCRIPTION
**Pull request by matborowczyk**
# Description

* Short description here *

closes *Add ticket reference here*

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [ ] Attached issue to pull request
- [ ] Changelog entry
- [ ] Code is clear and sufficiently documented
- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Correct, in line with design
~~- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )~~

# Reviewer Checklist:

- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Code is clear and sufficiently documented
- [x] Correct, in line with design

### CODE REVIEW

File: src/Slices/ServiceCatalog/UI/Tabs/AttributeTable.test.tsx [hunk 1]
```diff
@@ -83,3 +85,101 @@ test("GIVEN AttributeTable WHEN passed no attributes but some embedded entities
...
+    name: "Row-embedded$embedded_single$another_embedded$another_embedded_single",
+  });
+  expect(row).toBeVisible();
+  expect(row2).toBeVisible();
+  expect(row3).toBeVisible();
+
+  await userEvent.click(collapseAllButton);
+  expect(row).not.toBeVisible();
+  expect(row2).not.toBeVisible();
+  expect(row3).not.toBeVisible();
```
Comment (LauraStordeur) [inline src/Slices/ServiceCatalog/UI/Tabs/AttributeTable.test.tsx:184]: Missing t he test case when closing them back :)

Comment (matborowczyk) [reply src/Slices/ServiceCatalog/UI/Tabs/AttributeTable.test.tsx:184]: You mean, one by one?

File: src/UI/Components/ToggleAll/ToggleAll.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import React from "react";
+import { Button } from "@patternfly/react-core";
+import {
+  AngleDoubleDownIcon,
+  AngleDoubleUpIcon,
+} from "@patternfly/react-icons";
+
+interface ToggleAllProps {
+  isExpanding: boolean;
```
Comment (LauraStordeur) [inline src/UI/Components/ToggleAll/ToggleAll.tsx:9]: isExpanded (expanding would mean the action is pending)

File: src/UI/Components/TreeTable/Helpers/TreeExpansionManager.ts [hunk 3]
```diff
@@ -7,23 +7,52 @@ export interface ExpansionState {
...
     if (state[key]) return this.close(state, key);
     return this.open(state, key);
   }
 
+  toggleAll(
+    state: ExpansionState,
+    key: string,
+    shouldOpen: boolean
+  ): ExpansionState {
+    if (shouldOpen) return this.openAll(state, key);
```
Comment (LauraStordeur) [inline src/UI/Components/TreeTable/Helpers/TreeExpansionManager.ts:29]: blocked statement please ;)

File: src/UI/Components/TreeTable/Helpers/TreeExpansionManager.ts [hunk 4]
```diff
@@ -7,23 +7,52 @@ export interface ExpansionState {
...
+    return this.close(state, key);
+  }
+
   get(state: ExpansionState, key: string): boolean {
     return state[key];
   }
+  checkIfAllChildrenAreOpened(state: ExpansionState, key: string): boolean {
```
Comment (LauraStordeur) [inline src/UI/Components/TreeTable/Helpers/TreeExpansionManager.ts:36]: Are you still using this one somewhere? If not it can be removed I think :D

Comment (matborowczyk) [reply src/UI/Components/TreeTable/Helpers/TreeExpansionManager.ts:36]: Yeah, you right, I forgot to delete that one and one below right away :D

**Comment by bartv** [comment]
The icons is fine. Time will tell once we are using it.

**Review by LauraStordeur — APPROVED** [review]

**Pull request closed without merging**